enterprise/server/util/vsock/vsock.go

```diff
@@ -140,10 +140,12 @@ func dialHostToGuest(ctx context.Context, socketPath string, port uint32) (net.C
 	if n != len(fcConnectString) {
 		return nil, status.InternalErrorf("HostDial failed: wrote %d bytes, expected %d", n, len(fcConnectString))
 	}
+	conn.SetReadDeadline(time.Now().Add(1 * time.Second))
```

Comment:
Trying to make sure I understand how this works - does this sound right to you?

* Firecracker should start listening on v.sock as soon as the VM is started, even before we start the vmexec server, and should normally be able to reply to these CONNECT messages immediately
* When the CONNECT does get accepted by firecracker, but vmexec is not yet listening, firecracker will just terminate the connection (based on [this](https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md#host-initiated-connections)
* Sometimes we get unlucky though and firecracker doesn't start listening to v.sock by the time we call this func, so the Read() just hangs?
* This PR will make it so that if we happen to get unlucky, we'll just fail instead of waiting longer than we need to?

Also, in the case where firecracker terminates the connection (due to the vmexec server not yet listening) - I guess SimpleGRPCDial will retry? I see there's a backoff config but not 100% sure if the backoff is just retrying the bufDialer func or something else